tasks/pipeline.py

```diff
@@ -463,6 +466,65 @@ def trigger_child_pipeline(_, git_ref, project_name, variables="", follow=True):
         print("Child pipeline finished successfully")
 
 
+def parse(commit_str):
+    lines = commit_str.split("\n")
+    title = lines[0]
+    url = "NO_URL"
+    pr_id_match = re.search(r".*\(#(\d+)\)", title)
+    if pr_id_match is not None:
+        url = f"github.com/DataDog/datadog-agent/pull/{pr_id_match.group(1)}"
+    author = lines[1]
+    author_email = lines[2]
+    files = lines[3:]
+    return title, author, author_email, files, url  # Include author's email
+
+
+def is_system_probe(owners, files):
+    target = {("TEAM", "@DataDog/Networks"), ("TEAM", "@DataDog/universal-service-monitoring"),
+              ("TEAM", "@DataDog/ebpf-platform"), ("TEAM", "@DataDog/agent-security")}
+    for f in files:
+        match_teams = set(owners.of(f)) & target
+        if len(match_teams) != 0:
+            return True
+
+    return False
+
+
+@task
+def changelog(_, new_git_sha):
+    old_git_sha = sp.check_output(["git", "rev-list", "-n 1", "changelog-nightly-staging-sha"]).decode().strip()
+    commits = (
+        sp.check_output(
+            ["git", "log", f"{old_git_sha}..{new_git_sha}", "--pretty=format:%h"]
+        )
+        .decode()
+        .split("\n")
+    )
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Ditto (`sp.check_output` --> `ctx.run`)